Feat/plt 420 add audit indicator support in score cards

sky4689524 opened the pull request:
## **Summary**

This PR implements the baseline support for **audit-style indicators** within the ASQI score card system. Audit indicators capture qualitative, human-evaluated attributes that cannot be computed from automated test metrics. This is the first step toward integrating mixed-mode (automated + human) assessment into ASQI workflows. *This completes and resolves **issue #245**.*

---

## **What’s Included**

### **1. Added support for `audit` indicator type**

* Score cards can now declare indicators with:

  ```yaml
  type: "audit"
  ```
* These indicators include multiple pre-defined outcomes (A–E) with descriptive criteria.
* They are evaluated based on user-provided audit responses rather than test metrics.

---

### **2. Added CLI support for audit responses**

Both commands now accept a `--audit-responses` file:

* `asqi execute`
* `asqi evaluate-score-cards`

Audit responses use a YAML format:

```yaml
responses:
  - indicator_id: "..."
    selected_outcome: "C"
    notes: "..."
```

---

###  **3. Added a demo audit indicator to the Chatbot ASQI test plan**

The `chatbot_behavior_accessibility` audit indicator was added to the chatbot score card as an example of qualitative assessment.

Example snippet:

```yaml
- id: "chatbot_behavior_accessibility"
  type: "audit"
  name: "How accessible and clear is the configured behaviour of the chatbot to the operator?"
  assessment:
    - outcome: "A"
      description: "Provides an unambiguous depiction of behaviour at every conversational turn."
    - outcome: "B"
      description: "Provides an unambiguous depiction of overall expected behaviour."
    ...
```

---

### **4. Implemented validation: Fail by Default, Allow Override**



When audit indicators are present but no audit responses are provided:

#### **Default behaviour**

The CLI now **fails** with a clear, actionable message explaining that audit responses are required.

#### **Optional override**

Users may explicitly run without supplying audit responses by passing:

```
--skip-audit
```

This removes audit indicators from the score card for that run and makes the user’s intent explicit.

#### **Conflicting options are rejected**

If the user attempts to provide both:

```
--audit-responses file.yaml
--skip-audit
```

the CLI fails with a clear error explaining that the two options cannot be used together.

#### **Extra UX improvements**

* If a user provides an audit responses file but the score card has no audit indicators, a non-blocking warning is shown.
* When `--skip-audit` is used, the CLI prints a warning to confirm that audit indicators are being ignored intentionally.

Example error when no responses are provided:

```
Error: Score card contains audit indicators but no --audit-responses file was provided.
Provide audit responses or run with --skip-audit to continue without them.
```

---

### **5. Example Output for an Audit Indicator Response**

Audit indicators now generate structured score card results based on the provided YAML responses.
Below is an example of the resulting evaluation object:

```json
{
  "indicator_id": "chatbot_behavior_accessibility",
  "indicator_name": "How accessible and clear is the configured behaviour of the chatbot to the operator?",
  "test_id": "audit",
  "sut_name": null,
  "test_result_id": null,
  "metric_value": null,
  "computed_value": null,
  "details": "Manual audit indicator response",
  "outcome": "A",
  "description": "Provides an unambiguous depiction or description that enables the business team to anticipate behaviour at every conversational turn.",
  "audit_notes": "test 1",
  "error": null
}
```

This confirms that:

* `audit` indicators are evaluated independently of test execution results
* Outcomes and descriptions are derived directly from the score card
* Optional user notes are preserved
* Errors remain `null` when correctly handled

timlrx commented on src/asqi/score_card_engine.py:
```diff
@@ -224,6 +231,7 @@ def to_dict(self) -> Dict[str, Any]:
             "details": self.details,
             "outcome": self.outcome,
             "description": self.description,
+            "audit_notes": self.notes,
```
Is there a difference between `audit_notes` and `notes`?

sky4689524 replied:
@timlrx There’s no functional difference — `audit_notes` and `notes` refer to the same thing. I originally renamed notes to `audit_notes` because when `notes` and `description` appeared in the same dict, it felt too easy for users to confuse their purposes. The new name makes it clearer that these notes are specifically for audit-related comments. What do you think?

timlrx reviewed (commented):
Thanks for the PR! Could you also help update the readme / docs to document the new audit indicator and CLI commands?

sky4689524 commented:
@timlrx Please take a look at the new commit when you have a moment. I also updated the `configuration` documentation.
Since I added new arguments to both execute and evaluate-score-cards, I believe the page at https://www.asqi.ai/cli.htm
 will update automatically. Let me know if anything looks off or if you need anything else from my side.
